google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -3321,6 +3451,26 @@ private boolean canCreateSession() {
     }
   }
 
+  private void maybeCreateMultiplexedSession(SessionConsumer sessionConsumer) {
+    synchronized (lock) {
+      if (!multiplexedSessionBeingCreated) {
+        logger.log(Level.FINE, String.format("Creating multiplexed sessions"));
+        try {
+          multiplexedSessionBeingCreated = true;
+          sessionClient.createMultiplexedSession(sessionConsumer);
+        } catch (Throwable ignore) {
+          // such an exception will never be thrown. the exception will be passed onto the consumer.
+        }
+      }
+    }
+  }
+
+  private void handleCreateMultiplexedSessionFailure(SpannerException e) {
```

Comment:
nit: I think we should remove this method and just add the logic directly to the `onSessionCreateFailure` method of the initialization consumer.  As it is now, it might confuse future developers who believe that it is a bug that the maintainer consumer is not calling this method.

Reply:
Done.